pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -1183,6 +1168,10 @@ func (r ReconcileHyperConverged) applyUpgradePatch(req *common.HcoRequest, hcoJs
 		req.Logger.Info("applying upgrade patch", "knownHcoSV", knownHcoSV, "affectedRange", p.SemverRange, "patches", p.JSONPatch)
 		patchedBytes, err := p.JSONPatch.Apply(hcoJson)
 		if err != nil {
+			if errors.Cause(err) == jsonpatch.ErrTestFailed {
```

Comment:
Avoid considering an upgrade error a JSON Patch `test` operations used for data validation

Reply:
+1